tsk/auto/db_postgresql.cpp

```diff
@@ -1025,8 +1026,28 @@ int TskDbPostgreSQL::addFile(TSK_FS_FILE * fs_file, const TSK_FS_ATTR * fs_attr,
             return 1;
     }
 
-    char zSQL[2048];
-    snprintf(zSQL, 2048, "INSERT INTO tsk_files (fs_obj_id, obj_id, data_source_obj_id, type, attr_type, attr_id, name, meta_addr, meta_seq, dir_type, meta_type, dir_flags, meta_flags, size, crtime, ctime, atime, mtime, mode, gid, uid, md5, known, parent_path) "
+    char zSQL_fixed[2048];
+    zSQL_fixed[2047] = '\0';
+    char *zSQL_dynamic = NULL; // Only used if the query does not fit in the fixed length buffer
+    char *zSQL = zSQL_fixed;
+    int bufLen = 2048;
+
+    // Check if the path may be too long. The rest of the query should take up far less than 500 bytes.
+    if (strlen(name_sql) + strlen(escaped_path_sql) + 500 > bufLen) {
+        // The query may be long to fit in the standard buffer, so create a larger one.
+        // This should be a very rare case and allows us to not use malloc most of the time.
+        // The same buffer will be used for the slack file entry.
+        bufLen = strlen(escaped_path_sql) + strlen(name_sql) + 500;
+        if ((zSQL_dynamic = (char *)tsk_malloc(bufLen)) == NULL) {
+            PQfreemem(escaped_path_sql);
+            PQfreemem(name_sql);
+            return -1;
```

Comment:
Do we need to free name and escaped_path here as we do elsewhere?
Also, shouldn't this be returning 1 instead of -1?